subworkflows/local/virusbreakend_calling/main.nf

```diff
@@ -0,0 +1,154 @@
+    // channel: [ meta_virus, tumor_bam ]
+    ch_virusbreakend_inputs = ch_inputs_sorted.runnable
+        .map { meta, tumor_bam, tumor_bai ->
+
+            def meta_virus = [
+                key: meta.group_id,
+                id: meta.group_id,
+                sample_id: Utils.getTumorDnaSampleName(meta),
+            ]
+
+            return [meta_virus, tumor_bam]
+        }
+
+    // Run process
+    VIRUSBREAKEND(
+        ch_virusbreakend_inputs,
+        gridss_config,
+        genome_fasta,
+        genome_fai,
+        genome_dict,
+        genome_gridss_index,
+        virusbreakenddb,
+    )
+
+    ch_versions = ch_versions.mix(VIRUSBREAKEND.out.versions)
+
+    //
+    // MODULE: Virus Interpreter
+    //
+    // Select input sources
+    // channel: [ meta, virus_tsv, purple_dir, metrics ]
+    ch_virusinterpreter_inputs_selected = WorkflowOncoanalyser.groupByMeta(
+        WorkflowOncoanalyser.restoreMeta(VIRUSBREAKEND.out.tsv, ch_inputs),
+        ch_purple,
+        ch_bamtools_somatic,
+    )
+        .map { meta, virus_tsv, purple_dir, metrics ->
+
+            def inputs = [
+                virus_tsv,
+                Utils.selectCurrentOrExisting(purple_dir, meta, Constants.INPUT.PURPLE_DIR),
+                Utils.selectCurrentOrExisting(metrics, meta, Constants.INPUT.BAMTOOLS_TUMOR),
+            ]
+
+            return [meta, *inputs]
+        }
+
+    // Sort inputs
+    // channel: [ meta, virus_tsv, purple_dir, metrics ]
+    // channel: skip: [ meta ]
+    ch_virusinterpreter_inputs_sorted = ch_virusinterpreter_inputs_selected
+        .branch { meta, virus_tsv, purple_dir, metrics ->
+            runnable: virus_tsv && purple_dir && metrics
+            skip: true
+                return meta
+        }
+
+    // Create process input channel
+    // channel: [ meta_virus, virus_tsv, purple_dir, metrics ]
+    ch_virusinterpreter_inputs = ch_virusinterpreter_inputs_sorted.runnable
+        .map { d ->
+
+            def meta = d[0]
+            def inputs = d[1..-1]
+
+            def meta_virus = [
+                key: meta.group_id,
+                id: meta.group_id,
+                sample_id: Utils.getTumorDnaSampleName(meta),
+            ]
+
+            return [meta_virus, *inputs]
+        }
+
+    // Run process
+    VIRUSINTERPRETER(
+        ch_virusinterpreter_inputs,
+        virus_taxonomy_db,
+        virus_reporting_db,
+    )
+
+    ch_versions = ch_versions.mix(VIRUSINTERPRETER.out.versions)
+
+    // Set outputs, restoring original meta
+    // channel: [ meta, virusinterpreter_dir ]
+    ch_outputs = Channel.empty()
+        .mix(
+            WorkflowOncoanalyser.restoreMeta(VIRUSINTERPRETER.out.virusinterpreter_dir, ch_inputs),
+            ch_virusinterpreter_inputs_sorted.skip.map { meta -> [meta, []] },
+            ch_inputs_sorted.skip.map { meta -> [meta, []] },
+        )
+
+    emit:
+    virusinterpreter_dir = ch_outputs  // channel: [ meta, virusinterpreter_dir ]
+
```

Comment:
```suggestion
```

Reply:
Yes, I saw that you had this consistently, but github is so laggy reviewing this PR that I didn't want to scroll back to find which file I'd made the comment on! I have no objection to that.